ios/RPCModule.swift

```diff
@@ -48,31 +48,45 @@ class RPCModule: NSObject {
       }
   }
 
-  func saveWalletFile(_ data: String) {
-      let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
-      if let documentsDirectory = paths.first {
-        let fileName = "\(documentsDirectory)/wallet.dat.txt"
-        do {
-            try data.write(toFile: fileName, atomically: true, encoding: .utf8)
-        } catch {
-            NSLog("Error save wallet \(error.localizedDescription)")
+  func saveWalletFile(_ base64EncodedString: String) {
+      // need to decode the content first.
+      // save the decoded binary data
+      if let base64DecodedData = Data(base64Encoded: base64EncodedString) {
+        let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
+        if let documentsDirectory = paths.first {
+            let fileName = "\(documentsDirectory)/wallet.dat.txt"
+            do {
+              //NSLog("decoded data \(base64DecodedData)")
+              try base64DecodedData.write(to: URL(fileURLWithPath: fileName))
```

Comment:
Is this write operation defined to be atomic?   Is failure guaranteed to leave the original file untouched?

Reply:
We can always make our `saveWalletFile` atomic by writing to a second file `pre-wallet.dat.txt` and then only if that write operation succeeds we mv the old wallet out of the way, and move the new `pre-wallet.dat.txt` to `wallet.dat.txt`.  Then the previous version can be kept as `wallet.dat.txt-$timestamp` or something.   As long as wallet files are small we would then have this manual backup system in place.

Reply:
I don't noticed any problem in the `write` action so far, the problem was when the conversion from `b64 encoded binary` to `b64 encoded utf-8 string` result with a `nil` value. But the `nil` value was written correctly in the wallet file.